cura/Machines/Models/MaterialManagementModel.py

```diff
@@ -264,3 +270,52 @@ def removeFavorite(self, material_base_file: str) -> None:
             self.favoritesChanged.emit(material_base_file)
         except ValueError:  # Material was not in the favorites list.
             Logger.log("w", "Material {material_base_file} was already not a favorite material.".format(material_base_file = material_base_file))
+
+    def _onOutputDevicesChanged(self) -> None:
+        """
+        When the list of output devices changes, we may want to update the
+        preferred export path.
+        """
+        cura_application = cura.CuraApplication.CuraApplication.getInstance()
+        device_manager = cura_application.getOutputDeviceManager()
+        devices = device_manager.getOutputDevices()
+        for device in devices:
+            if device.__class__.__name__ == "RemovableDriveOutputDevice":
+                self._preferred_export_all_path = QUrl.fromLocalFile(device.getId())
+                break
+        else:  # No removable drives? Use local path.
+            self._preferred_export_all_path = cura_application.getDefaultPath("dialog_material_path")
+        self.outputDevicesChanged.emit()
+
+    outputDevicesChanged = pyqtSignal()  # Triggered when adding or removing removable drives.
+    @pyqtProperty(QUrl, notify = outputDevicesChanged)
+    def preferredExportAllPath(self) -> QUrl:
```

Comment:
Isn't it neater to just turn this into a `getPreferredExportAllPath() ` (so a slot) and only calculate it before the exportAllMaterialsDialog dialog is opened? 

It feels a bit strange that we re-calculate the path every time that a device is added (even if the user might not ever use this). 

If that is changed, you can also remove the extra code that you added in the init.

Reply:
Yeah, agreed. Basically the advantages are:

* No performance impact for people not using the sync button (=99.9% of users).
* Simplification of code, not requiring that extra signal to update listeners of the property.

Disadvantages are:

* Every time the dialogue is opened, we need to update the folder.

Since the dialogue is really only opened from this one button, and in the future from a pop-up instead but still only in one spot, the disadvantage is very minor. The advantages are better. I'll change it!